### PR TITLE
Documentation Change for Disable Cache

### PR DESCRIPTION
**Pull request by sra122**

### CODE REVIEW

**Comment by saeideng** [comment]
`withDisabledCache` or immutable methods at all need to reassign  or need to return immediately 
```php
//mutable
$this->response->x();
//immutable 
$this->response = $this->response->withX();

File: en/controllers/components/request-handling.rst [hunk 1]
```diff
@@ -128,6 +128,9 @@ However, you want to allow caching for non-AJAX requests. The
 following would accomplish that::
 
         if ($this->request->is('ajax')) {
+            
+            $this->response->withDisabledCache();
```
Comment (dereuromark) [inline en/controllers/components/request-handling.rst]: with..() and no assignment? That looks wrong.

Comment (garas) [inline en/controllers/components/request-handling.rst]:
```suggestion
            $this->response = $this->response->withDisabledCache();
```

File: en/controllers/components/request-handling.rst [hunk 2]
```diff
@@ -128,6 +128,9 @@ However, you want to allow caching for non-AJAX requests. The
 following would accomplish that::
 
         if ($this->request->is('ajax')) {
+
```
Comment (dereuromark) [inline en/controllers/components/request-handling.rst]: the newline is only noise.

**Comment by saeideng** [comment]
I pushed the new line to bottom https://github.com/cakephp/docs/commit/1b61d8d86

**Pull request merged**